component/mcsdadmin/component.go

```diff
@@ -0,0 +1,205 @@
+		return
+	}
+
+	props := struct {
+		Organizations []Organization
+	}{
+		Organizations: organizations,
+	}
+
+	w.WriteHeader(http.StatusOK)
+	RenderWithBase(w, "service_edit.html", props)
+}
+
+func ServiceNewPostHandler(w http.ResponseWriter, r *http.Request) {
+	log.Debug().Msg("New post for HealthcareService resource")
+
+	resourceType := "HealthcareService"
+
+	r.ParseForm()
+	var data = map[string]string{}
+	data["resourceType"] = resourceType
+	data["name"] = r.PostForm.Get("name")
+	data["active"] = r.PostForm.Get("active")
+	data["providedBy"] = r.PostForm.Get("providedBy")
+
+	_, err := ResourceFromMap(resourceType, data)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+
+	services, err := FindAllServices()
+	if err != nil {
+		log.Warn().Err(err).Msg("Failed to find all services")
+	}
+
+	RenderWithBase(w, "service_list.html", services)
+}
+
+func ServiceEditHandler(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusNotImplemented)
+	RenderWithBase(w, "service_edit.html", nil)
+}
+
+func OrganizationHandler(w http.ResponseWriter, r *http.Request) {
+	organizations, err := FindAllOrganizations()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	RenderWithBase(w, "organization_list.html", organizations)
+}
+
+func OrganizationNewHandler(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+	RenderWithBase(w, "organization_edit.html", nil)
+}
+
+func OrganizationNewPostHandler(w http.ResponseWriter, r *http.Request) {
+	log.Debug().Msg("New post for organization resource")
+
+	resourceType := "Organization"
+
+	r.ParseForm()
+	var data = map[string]string{}
+	data["resourceType"] = resourceType
+	data["name"] = r.PostForm.Get("name")
+	data["active"] = r.PostForm.Get("active")
+
+	_, err := ResourceFromMap(resourceType, data)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+
+	organizations, err := FindAllOrganizations()
+	if err != nil {
+		log.Warn().Err(err).Msg("Failed to find all organizations")
+	}
+
+	RenderWithBase(w, "organization_list.html", organizations)
+}
+
+func HomePageHandler(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+	RenderWithBase(w, "home.html", nil)
+}
+
+func NotImplementedHandler(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusNotImplemented)
+	RenderWithBase(w, "home.html", nil)
+}
+
+func NotFoundHandler(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusNotFound)
+	_, _ = w.Write([]byte("Path not implemented"))
+}
+
+func (c Component) RegisterHttpHandlers(mux *http.ServeMux) {
+	mux.HandleFunc("/mcsdadmin/healthcareservice", ServiceHandler)
+	mux.HandleFunc("/mcsdadmin/healthcareservice/new", ServiceNewHandler)
```

Comment:
can you make the handler names more concise? Maybe something, simply, like `newService`, `listOrganizations`, `editOrganization` etc. I don't think you need the "Handler" postfix, although you can add it if you like them.

Reply:
Must have been all that C# I've been writing that got me into the habit of long names.